### ISSUE TITLE
DateRangePicker select Year/Month

### ISSUE THREAD

**Issue by jinqiangg** [mui/mui-x#4546]
Correct me if I miss out anything. While using DateRangePicker on v5, I am unable to do a quick selection to change year / month. Have to manually click on the < or > button. Understand that the range design with the intend of selecting recent dates but I do have use case that requires users to select date from years ago.

- [X] I have searched the [issues](https://github.com/mui-org/material-ui/issues) of this repository and believe that this is not a duplicate.

## Summary 💡

Allow selection of year or month upon clicking the month / year.

**Comment by oliviertassinari** [comment]
> It would be maybe also nice to have a MonthRangePicker and YearRangePicker for the time being I will use tow DatePickers with month/year views

@socramm9 Is this #4995 what you are referring to?

**Comment by socramm9** [comment]
> @socramm9 Is this #4995 what you are referring to?

Yes I build something similar myself now. It is not perfect and has some other issues but here some Images for Inspiration
<img width="460" alt="Bildschirmfoto 2022-05-29 um 21 54 11" src="https://user-images.githubusercontent.com/52098754/170889120-3c28455f-0f80-459d-88da-4ca5a28e3bf6.png">
<img width="458" alt="Bildschirmfoto 2022-05-29 um 21 54 02" src="https://user-images.githubusercontent.com/52098754/170889122-093e5fd3-df2f-4463-b4aa-73d30ad6e258.png">
<img width="727" alt="Bildschirmfoto 2022-05-29 um 21 53 42" src="https://user-images.githubusercontent.com/52098754/170889124-f7a204f2-6979-480d-b94f-392a3a2baf83.png">

**Comment by vazkir** [comment]
> > @socramm9 Is this #4995 what you are referring to?
> 
> Yes I build something similar myself now. It is not perfect and has some other issues but here some Images for Inspiration <img alt="Bildschirmfoto 2022-05-29 um 21 54 11" width="460" src="https://user-images.githubusercontent.com/52098754/170889120-3c28455f-0f80-459d-88da-4ca5a28e3bf6.png"> <img alt="Bildschirmfoto 2022-05-29 um 21 54 02" width="458" src="https://user-images.githubusercontent.com/52098754/170889122-093e5fd3-df2f-4463-b4aa-73d30ad6e258.png"> <img alt="Bildschirmfoto 2022-05-29 um 21 53 42" width="727" src="https://user-images.githubusercontent.com/52098754/170889124-f7a204f2-6979-480d-b94f-392a3a2baf83.png">

Do you by any chance have any example on how you achieved this?

**Comment by socramm9** [comment]
@vazkir  Yes it not up to MUI standards but its usable. Fell free to modify it however you want 

https://codesandbox.io/s/busy-babycat-96jksl

**Comment by cube-dan** [comment]
@vazkir, [You're my hero](https://github.com/mui/mui-x/issues/4546#issuecomment-1189971012) ... pulled it down, turned it into a component, can’t stop smiling. 

Thanks SO much for posting your solution!

**Comment by plgagnon2** [comment]
Particularly need this for analytics application...

**Comment by bruns6077** [comment]
https://material.angular.io/components/datepicker/overview

does handle this also pretty well, might be something for inspiration

**Comment by LukasTy** [comment]
We agreed to first add support for changing `year` and `month` views in a similar way to how Angular Material does it.

With a limitation of only supporting it when `calendars={1}`.
If the `calendars` prop is not overridden by the user, we would default it to `1` if `views` has `year` or `month` entries.
Here is the issue for enhancing the behavior to add the missing behavior: https://github.com/mui/mui-x/issues/14119.

This does not impact anything related to picking the month and year ranges.

**Comment by gabbi-pf** [comment]
Hello, we reported [#14874](https://github.com/mui/mui-x/issues/14874), sorry for the duplicate but we couldn't find this one when we searched for issues. Should this now be marked with `plan: premium`?
Also, is there an estimate on when support for `year` and `month` view will be supported? I see this issue being opened 3 years ago, and we will need to find another solution if this functionality isn't supported in the near future. We understand you get a million requests like these and have to prioritize, but we need to evaluate other solutions if this is not implemented soon. 🫤

> Duplicates #4546
> 
> You can find this issue to track our advancement on the topic, this is not yet supported but we plan to add support for it at some point

**Comment by vaneri** [comment]
@LukasTy do you think that you could help on this particular topic? / it would be great to know if there is at least a plan for it
thanks!

**Comment by flaviendelangle** [comment]
@gabbi-pf I added the requested label :+1: 

We don't plan to support it in the new feature because we have a very heavy roadmap for the coming months.

**Comment by DimitarKazakov** [comment]
I managed to make a workaround using DateRangeCalendar for the range selection and DateCalendar for the year and month selection. I am posting part of my code here if anyone still needs the component. Ignore the styles and the customization, the important part to look at is the monthYearView state, the dateRangeCalendarProps ref and how i am using DateCalendar and DateRangeCalendar with the custom calendarHeader. PS: i am new to posting in github discussions

https://github.com/user-attachments/assets/a7dfa897-c8da-433e-9a9d-ba11a801e0f3

[code.txt](https://github.com/user-attachments/files/18621136/example.txt)

**Comment by LukasTy** [comment]
This would be mostly covered by https://github.com/mui/mui-x/issues/9124.
But we plan to implement this only on the new DX.

**Comment by RockPaperGardenHose** [comment]
@LukasTy  When can we expect this addition to the range pickers? have you planned when you will start implementing it/release it ? :)

**Comment by saarbyrner** [comment]
Putting emphasis on this too. I think there are 2 tickets for this, I suppose that show how important it is.

**Comment by RockPaperGardenHose** [comment]
@saarbyrner Ok, nice. I see that this issue and the other one you mention have been in "next in queue" since probably more than a year ago. Will this issue be in progress anytime soon ?

**Comment by HristoP96** [comment]
I am using premium version of MUI and I am really disappointed that this feature is not included out of the box for the date range picker. 

I am having the case where users should be able to pick a start date that is years behind the current one, and to do that with what is currently available as a date range picker they need to click multiple times to go to the desired date...

Given the fact that such month views and year views exist in the MUI toolkit, the severity for this feature to be included in soon should be high.

**Comment by zac1wz** [comment]
Our team is paying for **Premium** MUI license and we are very disappointed that this feature is not implemented after **4 years**.

We are waiting for this to be implemented soon, as we are having next use case that needs to be implemented.

**Comment by oliviertassinari** [comment]
> we are having next use case that needs to be implemented.

I'm curious about this use case. Do you have more details? Thanks

**Comment by azamatguseinov** [comment]
Same here, we are pro/premium users.
Almost all of our use-cases for RangePicker are multi-year selection.
Current RangePicker is close to unusable without it for us. 

Would appreciate at least clarification on whether it is planned to be implemented or not in near future. 
So we could plan our roadmap and implement the feature ourselves in case if no near term plans on it. 

Thanks!

**Comment by LukasTy** [comment]
Thank you for expressing your interest in this feature. 🙏 
While we are still waiting for the built-in feature, what do you think about a calendar header with year navigation in addition to the existing month navigation?
Here is a minimal example: https://stackblitz.com/edit/3fquplch?file=src%2FDemo.tsx.

**Comment by AidanAtSigma2** [comment]
@LukasTy @oliviertassinari Our use cases are for multi-year and multi-month selections. This seems like a very intuitive use case for a date range picker. The DateCalendar works perfectly with its `views` property, and we assumed the date range picker would have the same capability when we purchased pro. We are really hoping to invest more in MUI X - what's the expected release date for this feature? Having to build our own custom date range component using the DateCalendar (just for its views feature), or a custom header for the date range component, may force us to consider an alternative.

**Comment by SamuelFornerJurisConcept** [comment]
Hi,
Any update about the year navigation on the range picker ?
@LukasTy  also it seems that your link is not working
thank you !

**Comment by LukasTy** [comment]
Hello @AidanAtSigma2 and @SamuelFornerJurisConcept. Thank you for reminding me about this feature.
Sadly, I currently don't have any specific ETA for when this will be ready.

> also it seems that your link is not working

Do you mean https://stackblitz.com/edit/3fquplch?file=src%2FDemo.tsx?
I have tried it in multiple browsers and it works without issues. 🤷 🙈

**Comment by SamuelFornerJurisConcept** [comment]
> Hello [@AidanAtSigma2](https://github.com/AidanAtSigma2) and [@SamuelFornerJurisConcept](https://github.com/SamuelFornerJurisConcept). Thank you for reminding me about this feature. Sadly, I currently don't have any specific ETA for when this will be ready.
> 
> > also it seems that your link is not working
> 
> Do you mean https://stackblitz.com/edit/3fquplch?file=src%2FDemo.tsx? I have tried it in multiple browsers and it works without issues. 🤷 🙈

Hello,
The link is working, sorry, it was because of the global cloudflare issue.

**Comment by xantos008** [comment]
Is any progress on the issue?

**Comment by michelengelen** [comment]
Thanks everyone for the detailed feedback and use cases here.

To be transparent about where this stands in the code: views/view on DateRangeCalendar and the range picker components are still hard-typed to 'day' only (DateRangeCalendar.types.ts; DesktopDateRangePicker.tsx and MobileDateRangePicker.tsx both carry // TODO: ... once we support month and year views), and the range calendar only ever renders the day grid at runtime. The two follow-up issues that would unlock this, #9124 (header redesign) and #14119 (calendars > 1 support), are also still open with no PR attached.

In the meantime, the calendarHeader slot is public API today and gets you fast year-level paging without a full year/month grid:

```tsx
<DateRangeCalendar
  calendars={1}
  slots={{ calendarHeader: CustomCalendarHeader }}
/>
```

See docs/data/date-pickers/custom-components/CalendarHeaderComponent.tsx for the double-chevron year-navigation pattern; it only needs onMonthChange(currentMonth.add(1, 'year')) wired to your own buttons.

We know this is a long-standing, heavily requested feature and don't want to hand out another vague ETA. Keeping this open.
